### ISSUE TITLE
CURLOPT_MAXAGE_CONN value seems to be ignored

### ISSUE THREAD

**Issue by baranyaib90** [curl/curl#15856]
### I did this

I have set CURLOPT_MAXAGE_CONN value to 300L on a curl easy handle.

### I expected the following

I was expecting, that connections will be reused within 300 seconds, but instead I am seeing this in the logs:

```
[D] 1735452097.003733 https_client.c:214 C0D1: * Too old connection (211 seconds idle), disconnect it
[D] 1735452387.304260 https_client.c:214 D941: * Too old connection (280 seconds idle), disconnect it
[D] 1735452677.614517 https_client.c:214 38BE: * Too old connection (202 seconds idle), disconnect it
[D] 1735453336.182739 https_client.c:214 BB0B: * Too old connection (211 seconds idle), disconnect it
[D] 1735453628.973958 https_client.c:214 8850: * Too old connection (289 seconds idle), disconnect it
[D] 1735454215.510075 https_client.c:214 CC26: * Too old connection (486 seconds idle), disconnect it
[D] 1735454215.512702 https_client.c:214 CC26: * Too old connection (586 seconds idle), disconnect it
[D] 1735454528.745111 https_client.c:214 CBD7: * Too old connection (209 seconds idle), disconnect it
[D] 1735454727.456226 https_client.c:214 745D: * Too old connection (198 seconds idle), disconnect it
[D] 1735455136.246010 https_client.c:214 6A0D: * Too old connection (189 seconds idle), disconnect it
[D] 1735455428.755813 https_client.c:214 1B2A: * Too old connection (125 seconds idle), disconnect it
[D] 1735455700.214284 https_client.c:214 42F0: * Too old connection (139 seconds idle), disconnect it
[D] 1735455970.761525 https_client.c:214 5F96: * Too old connection (268 seconds idle), disconnect it
[D] 1735455970.761768 https_client.c:214 5F96: * Too old connection (270 seconds idle), disconnect it
[D] 1735456329.035982 https_client.c:214 8AED: * Too old connection (178 seconds idle), disconnect it
[D] 1735456527.534475 https_client.c:214 C1B4: * Too old connection (198 seconds idle), disconnect it
[D] 1735456527.534720 https_client.c:214 C1B4: * Too old connection (198 seconds idle), disconnect it
[D] 1735456936.276973 https_client.c:214 840D: * Too old connection (148 seconds idle), disconnect it
[D] 1735457228.595764 https_client.c:214 6DC6: * Too old connection (286 seconds idle), disconnect it
[D] 1735457800.736249 https_client.c:214 58AF: * Too old connection (342 seconds idle), disconnect it
```

Obviously values like 486, 586 and 342 are correct, but most values are under 300.

### curl/libcurl version

curl 8.9.1 (x86_64-pc-linux-gnu) libcurl/8.9.1 OpenSSL/3.3.1 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.2 libssh2/1.11.0 nghttp2/1.62.1 librtmp/2.3 OpenLDAP/2.6.8
Release-Date: 2024-07-31, security patched: 8.9.1-2ubuntu2.2
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

### operating system

Ubuntu 24.10

### Extra information

- More curl logs: [flightrecorder.log](https://github.com/user-attachments/files/18270906/flightrecorder.log)
- Software that is using libcurl: https://github.com/aarond10/https_dns_proxy/blob/master/src/https_client.c#L314

**Comment by jay** [comment]
The check is pretty straightforward

https://github.com/curl/curl/blob/83bedbd730d62b83744cc26fa0433d3f6e2e4cd6/lib/url.c#L726-L735

Modify the printf to show the value of data->set.maxage_conn. My guess is it will be the default 118. You will have to investigate why 300 wasn't set.

**Comment by jay** [comment]
found [the problem](https://github.com/aarond10/https_dns_proxy/blob/484bd153bb85a51df1c5bede1b091be76537e0a7/src/https_client.c#L310-L315). CURLOPT_MAXAGE_CONN is an enum not a define so your code is skipped
~~~c
#ifdef CURLOPT_MAXAGE_CONN
  ASSERT_CURL_EASY_SETOPT(ctx, CURLOPT_TCP_KEEPALIVE, 1L);
  ASSERT_CURL_EASY_SETOPT(ctx, CURLOPT_TCP_KEEPIDLE, 50L);
  ASSERT_CURL_EASY_SETOPT(ctx, CURLOPT_TCP_KEEPINTVL, 50L);
  ASSERT_CURL_EASY_SETOPT(ctx, CURLOPT_MAXAGE_CONN, 300L);
#endif
~~~

**Comment by baranyaib90** [comment]
Hi @jay, thank you very much!
I'm ashamed that I was not paying attention that it is an enum...
Looks like I should have tested that part much better.
I guess instead of faulty define I should find a better way to check if CURLOPT_MAXAGE_CONN option is available or not.

**Comment by jay** [comment]
> I guess instead of faulty define I should find a better way to check if CURLOPT_MAXAGE_CONN option is available or not.

[CURLOPT_MAXAGE_CONN](https://curl.se/libcurl/c/CURLOPT_MAXAGE_CONN.html#AVAILABILITY) is available since 7.65.0 so you could change it to this:

~~~c
#if LIBCURL_VERSION_NUM >= 0x074100
ASSERT_CURL_EASY_SETOPT(ctx, CURLOPT_MAXAGE_CONN, 300L);
#endif
~~~
